lib/core/src/lib/datatable/data/object-datatable-adapter.ts

```diff
@@ -121,35 +121,33 @@ export class ObjectDataTableAdapter implements DataTableAdapter {
 
         if (sorting?.key) {
             this._rows.sort((a: DataRow, b: DataRow) => {
-                let left = a.getValue(sorting.key);
-                let right = b.getValue(sorting.key);
-
-                if (typeof left === 'number' && typeof right === 'number') {
-                    return sorting.direction === 'asc' ? left - right : right - left;
-                } else {
-                    if (left) {
-                        left = left instanceof Date ? left.valueOf().toString() : left.toString();
-                    } else {
-                        left = '';
-                    }
+                let left = a.getValue(sorting.key) ?? '';
+                let right = b.getValue(sorting.key) ?? '';
 
-                    if (right) {
-                        right = right instanceof Date ? right.valueOf().toString() : right.toString();
-                    } else {
-                        right = '';
-                    }
+                if (typeof left !== 'string') {
+                    left = left.valueOf().toString();
+                }
 
-                    return sorting.direction === 'asc' ? left.localeCompare(right) : right.localeCompare(left);
+                if (typeof right !== 'string') {
+                    right = right.valueOf().toString();
                 }
+
+                return sorting.direction === 'asc'
+                    ? left.localeCompare(right, 'en', sorting.options)
```

Comment:
In this case instead of using host's local you'll always use `en` local, is it fine for us?

Reply:
in most cases it is safe, but that could break a little with special letters

Reply:
I think we should discuss this with Agnieszka and Denys to make sure we're not introducing breaking changes to the datatable